jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/InsightsUtils.kt

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResultField
+
+/**
+ * Mapped from response from the GetQueryResults call
+ */
+typealias LogResult = Map<String, String>
+/**
+ * Returned from the GetLogRecordResponse call
+ */
+typealias LogRecord = Map<String, String>
+typealias LogRecordFieldPair = Pair<String, String>
+
+fun List<ResultField>.toLogResult() = this.map { it.field() to it.value() }.toMap()
+
+fun LogResult.ptr(): String = this["@ptr"] ?: throw IllegalStateException("CWL GetQueryResults returned record without @ptr field")
```

Comment:
I was really confused what this meant seeing it in context, could we call it `identifier` or `id`. As is, it seems like something that would call JNI.